ocs_ci/utility/azure_utils.py

```diff
@@ -461,6 +475,281 @@ def get_storage_account_properties(self, storage_account_name):
+        base_domain = config.ENV_DATA["base_domain"]
+        self.create_network()
+        cmd = (
+            f"az aro create --resource-group {resource_group} --cluster-resource-group {cluster_resource_group} "
+            f"--name {cluster_name} --version {ocp_version} --vnet {vnet} --master-subnet {master_subnet} "
+            f"--worker-subnet {worker_subnet} --pull-secret @{pull_secret_path} "
+            f"--worker-vm-size {worker_flavor} --master-vm-size {master_flavor}  "
+            f"--worker-count {worker_replicas} --domain {cluster_name}.{base_domain}"
+        )
+        logger.info("Creating Azure ARO cluster.")
+        out = exec_cmd(cmd, timeout=3600).stdout
+        self.set_dns_records(cluster_name, resource_group, base_domain)
+        logger.info(f"Cluster deployed: {out}")
+        cluster_info = self.get_cluster_details(cluster_name)
+        # Create metadata file to store the cluster name
+        cluster_info["clusterName"] = cluster_name
+        cluster_info["clusterID"] = cluster_info["id"]
+        cluster_path = config.ENV_DATA["cluster_path"]
+        metadata_file = os.path.join(cluster_path, "metadata.json")
+        with open(metadata_file, "w+") as f:
+            json.dump(cluster_info, f)
+        installer_log_file = os.path.join(cluster_path, ".openshift_install.log")
+        formatted_time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        cluster_address = (
+            f"http://console-openshift-console.apps.{cluster_name}.{base_domain}"
+        )
+        logger.info(f"Cluster URL: {cluster_address}")
+        with open(installer_log_file, "w+") as f:
+            f.writelines(
+                [
+                    "W/A for our CI to get URL to the cluster in jenkins job. "
+                    "Cluster is deployed via az aro command!\n"
+                    f'time="{formatted_time}" level=info msg="Access the OpenShift web-console here: '
+                    f"{cluster_address}\"\n'",
+                ]
+            )
+        self.get_kubeconfig(cluster_name, resource_group)
+        self.write_kubeadmin_password(cluster_name, resource_group)
+        configure_ingress_and_api_certificates(skip_tls_verify=True)
+
+    def create_network(self):
+        """
+        Create network related stuff for the cluster.
+        """
+        vnet = config.ENV_DATA.get("aro_vnet", "aro-vnet")
+        vnet_address_prefixes = config.ENV_DATA.get(
+            "aro_vnet_address_prefixes", "10.0.0.0/22"
+        )
+        master_subnet = config.ENV_DATA.get("aro_master_subnet", "master-subnet")
+        master_subnet_address_prefixes = config.ENV_DATA.get(
+            "aro_master_subnet_address_prefixes", "10.0.0.0/23"
+        )
+        worker_subnet = config.ENV_DATA.get("aro_worker_subnet", "worker-subnet")
+        worker_subnet_address_prefixes = config.ENV_DATA.get(
+            "aro_worker_subnet_address_prefixes", "10.0.2.0/23"
+        )
```

Comment:
not a stopper.  could we make subnets as constants in defaults.py

Reply:
Done